src/index.js

```diff
@@ -0,0 +1,168 @@
+import './style.css';
+import {
+  getMoviesData,
+  postLikes, postComment,
+  getLikesForUnclick,
+  fetchCommentsFromApi,
+  renderComments,
+  getLikes,
+} from './modules/functionalities.js';
+import { showApiUrl } from './modules/showsAPI.js';
+
+// api imports
+import { likeApi, commentApi } from './modules/involvementAPI.js';
+import 'bootstrap';
+
+// testing
+import { countComments } from './modules/counter.js';
+
+import './assets/bg-for-page.jpg';
+import './assets/button-menu.png';
+import './assets/close-icon.svg';
+import modalContent from './modules/reservation';
+
+const renderMovies = async () => {
+  const data = await getMoviesData(showApiUrl);
+  data.sort(() => 0.5 - Math.random());
+  data.length = 20;
+  const movieContainer = document.getElementById('movieContainer');
+
+  data.forEach(async (movie) => {
+    const movieCard = document.createElement('div');
+    movieCard.classList.add('col-md-3', 'col-sm-6', 'mb-4');
+    movieCard.innerHTML = `
+      <div class="card custom-card">
+        <img src=${movie.image.medium} class="card-img-top" alt="images">
+        <div class="card-body">
+          <div  class="name-like-button">
+             <div>
+              <span class="card-title">${movie.name}</span>
+             </div>
+            <div  class="likeBtnContainer">
+              <span class="likesCount${movie.id}">0</span>
+              <span class="likeBtn" data-movie-id="${movie.id}">&#9825</span>
+            </div>
+          </div>
+          <div class="card-button">
+            <button type="button" 
+                    class="btn   btn-small comment-button " 
+                    data-bs-toggle="modal" 
+                    data-bs-target="#commentModal-${movie.id}">
+              Comment
+            </button>
+
+            <button type="button"  
+                    class="btn  reservationBtn"  
+                    id="${movie.id}" data-toggle="modal" data-target="#exampleModal-${movie.id}" >Reservations</button>
+          </div>
+        </div>
+      </div>
+    `;
+
+    movieContainer.appendChild(movieCard);
+
+    //  likes functionality
+    const likeBtn = movieCard.querySelector('.likeBtn'); // Select the like button within the movie card
+    let isLiked = false;
+    likeBtn.addEventListener('click', async () => {
+      if (isLiked) {
+        // If already liked, subtract the like and change the symbol back
+        likeBtn.innerHTML = '&#9825';
+        isLiked = false;
+        await getLikesForUnclick(movie.id);
+      } else {
+        // If not liked, add the like and change the symbol to ❤
+        likeBtn.innerHTML = '❤️';
+        isLiked = true;
+        await postLikes(movie.id, likeApi);
+        await getLikes(movie.id);
+      }
+    });
+    // update UI on page load
+    await getLikes(movie.id);
+
+    // modal functionality
+    const modal = document.createElement('div');
```

Comment:
- [ ] Currently, upon the initial loading of the page, a modal for movie comments is being created for each movie and data is retrieved for all movies at that point. This approach has some drawbacks, as it retrieves comments for all movies in one go, which might not be efficient. As per the project requirement, when a user clicks on the "Comments" button, the Comments popup should appear, and the web app should retrieve data only for the selected movie from the API. This ensures that details about the selected item are shown in the Comments popup, rather than loading data for all items during the initial movie load.

![image](https://github.com/CobbyElsonfx/JavaScript-Capstone/assets/55840999/13cd996f-b6ee-4b53-b8f1-157f1475e742)

Please ensure that your implementation adheres to this requirement, as it will help optimize the performance and user experience of your web app. Keep up the good work! 🌟📝🚀"